plugins/discovery/discovery_test.go

```diff
@@ -1276,3 +1357,527 @@ func TestInterQueryBuiltinCacheConfigUpdate(t *testing.T) {
 		t.Fatalf("Expected cache max size bytes to be 200 after discovery reconfigure, got: %v", config2.InterQueryBuiltinCache.MaxSizeBytes)
 	}
 }
+
+func TestPluginManualTriggerLifecycle(t *testing.T) {
```

Comment:
@gshively11 does this test capture the sequence of event you had in mind for your use-case ?

Reply:
I think so, although it's kind of hard to tell. I'll probably have to just try implementing your branch and see how it goes with some of our sample projects.